src/models/flight_control/FGLinearActuator.cpp

```diff
@@ -0,0 +1,252 @@
+        rate = 0.3;
+        minRate = -0.3;
+        maxRate = 0.3;
+        lag = 0.0;
+        lagPrevius = 0.0;
+        gain = 1.0;
+        
+        if (element->FindElement("set")) {
+            setProperty = PropertyManager->CreatePropertyObject<double>(element->FindElementValue("set"));
+            isSetProperty = true;
+        }
+        if (element->FindElement("reset")) {
+            resetProperty = PropertyManager->CreatePropertyObject<double>(element->FindElementValue("reset"));
+            isResetProperty = true;
+        }
+
+        if (element->FindElement("versus")) versus = element->FindElementValueAsNumber("versus");
+        if (element->FindElement("lag")) lag = element->FindElementValueAsNumber("lag");
+        if (element->FindElement("module")) module = element->FindElementValueAsNumber("module");
+        if (element->FindElement("rate")) {
+            rate = element->FindElementValueAsNumber("rate");
+            if (rate <= 0 or rate > 1.0) rate = 0.5;
+            minRate = -(rate);
+            maxRate = rate;
+        }
+        if (element->FindElement("gain")) gain = element->FindElementValueAsNumber("gain");
+        
+        FGFCSComponent::bind();
+        
+        Debug(0);
+    }
+    
+    //%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+    
+    FGLinear_Actuator::~FGLinear_Actuator()
+    {
+        Debug(1);
+    }
+    
+    //%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+    
+    bool FGLinear_Actuator::Run(void )
+    {
+        Input = InputNodes[0]->getDoubleValue();
+        if (initialized) {
+            input_prec = Input;
+            initialized = 0;
+            std::cerr << "FGLinear_Actuator Cicle_zero " << InputNodes[0]->GetNameWithSign() << std::endl;
+            return true;
+        }
+        
+        Output = 0.0;
+        
+        set = 1.0;
+        reset = 0.0;
+        
+        if (isSetProperty) set = setProperty;
+        if (isResetProperty) reset = resetProperty;
+            
+        if (set > 0 ) {
+            
+            double input_delta = Input - input_prec;
+            
+            if ((versus < 0.0 and input_delta > 0.0) or (versus  > 0.0 and input_delta < 0.0)) {
+                input_delta = 0.0;
+            }
+            input_prec = Input;
+            
+            if (input_delta != 0.0) {
+                if (input_delta <= (module * maxRate) and input_delta >= (module * minRate)) {
+                    if (input_delta > 0.0) {
+                        direction = 1;
+                        //cout << "FGLinear_Actuator::Run" << InputNodes[0]->GetNameWithSign() << " direction + " << direction << endl;
+                    } else if (input_delta < 0.0) {
+                        direction = -1;
+                        //cout << "FGLinear_Actuator::Run" << InputNodes[0]->GetNameWithSign() << " direction - " << direction << endl;
+                    }
+                }
```

Comment:
Since `input_delta` is non zero, `direction` can be initialized with the sign of `input_delta` instead of the block between lines 131-138 : 
```c++
        direction = sign(input_delta);
```
You might need to include `FGJSBBase.h` for the `sign` function to be recognized by the compiler.